src/scene.rs

```diff
@@ -1,33 +1,58 @@
-use std::collections::HashMap;
-use std::mem;
-use std::sync::Arc;
-
-use crate::callback;
-use crate::device::Device;
-use crate::geometry::Geometry;
-use crate::intersect_context::IntersectContext;
-use crate::ray::{Ray, RayHit};
-use crate::ray_packet::{Ray4, RayHit4};
-use crate::ray_stream::{RayHitN, RayN};
-use crate::sys::*;
+use crate::{Error, SceneFlags};
+use std::{collections::HashMap, mem, sync::Arc};
+
+use crate::{
+    callback,
+    device::Device,
+    geometry::Geometry,
+    intersect_context::IntersectContext,
+    ray::{Ray, RayHit},
+    ray_packet::{Ray4, RayHit4},
+    ray_stream::{RayHitN, RayN},
+    sys::*,
+};
 
 /// A scene containing various geometry for rendering. Geometry
 /// can be added and removed by attaching and detaching it, after
 /// which the scene BVH can be built via `commit` which will
 /// return a `CommittedScene` which can be used for ray queries.
 pub struct Scene {
     pub(crate) handle: RTCScene,
-    pub(crate) device: Arc<Device>,
+    pub(crate) device: Device,
     geometry: HashMap<u32, Arc<dyn Geometry>>,
```

Comment:
Maybe we can't remove the Arc around the geometry here because we're using dynamic dispatch/trait objects?